### PR TITLE
Updates to Integration tests to pass against Alpine

### PR DESCRIPTION
**Pull request by sivel**
##### SUMMARY
Updates to Integration tests to pass against Alpine

This depends on a new Alpine distro-test-container that I haven't sent a PR for yet.

##### ISSUE TYPE
- Feature Pull Request

##### COMPONENT NAME
many

##### ADDITIONAL INFORMATION

```paste below

```

### CODE REVIEW

File: test/integration/targets/subversion/roles/subversion/tasks/setup.yml [hunk 1]
```diff
@@ -11,6 +11,11 @@
   package:
     name: '{{ subversion_packages }}'
     state: present
+  when: ansible_distribution != 'Alpine'
+
+- name: install SVN pre-reqs - Alpine
+  command: 'apk add -U {{ subversion_packages|join(" ") }}'
```
Comment (sivel) [inline test/integration/targets/subversion/roles/subversion/tasks/setup.yml:17]: We don't have the `apk` module here, and as far as tests are concerned, I think this is sufficient.

File: test/integration/targets/handler_race/roles/random_sleep/tasks/main.yml [hunk 2]
```diff
@@ -2,7 +2,7 @@
 # tasks file for random_sleep
 - name: Generate sleep time
   set_fact:
-    sleep_time: "{{ 60 | random }}"
+    sleep_time: "{{ 30 | random }}"
```
Comment (sivel) [inline test/integration/targets/handler_race/roles/random_sleep/tasks/main.yml:5]: This isn't related to any fix, but when running tests I noticed these tests were pretty slow.

**Comment by mattclay** [comment]
@sivel The `quay.io/ansible/alpine3-test-container:1.18.0` image is now available. Can you update this PR?

File: test/lib/ansible_test/_data/completion/docker.txt [hunk 3]
```diff
@@ -1,3 +1,4 @@
+alpine3 name=quay.io/ansible/alpine3-test-container:1.18.0 python=3.8
```
Comment (mattclay) [inline test/lib/ansible_test/_data/completion/docker.txt]: I've been keeping this file sorted, except for the default entries being at the top, so I'd put this just before centos6.

**Comment by mattclay** [comment]
Once this is merged we should announce the availability of the alpine container here: https://github.com/ansible-collections/overview/issues/45

File: test/integration/targets/connection_ssh/runme.sh [hunk 4]
```diff
@@ -22,20 +22,22 @@ if command -v sshpass > /dev/null; then
             -e ansible_sshpass_prompt=notThis: \
             -e ansible_password=foo \
             -e ansible_user=definitelynotroot \
-	    -i test_connection.inventory \
+        -i test_connection.inventory \
```
Comment (mattclay) [inline test/integration/targets/connection_ssh/runme.sh]: It looks like the indentation is off here:

```suggestion
            -i test_connection.inventory \
```

**Review by mattclay — CHANGES_REQUESTED** [review]

File: test/integration/targets/setup_cron/tasks/main.yml [hunk 5]
```diff
@@ -19,17 +19,23 @@
...
-    - name: install cron and faketime packages
+    - name: install faketime packages
       package:
         name: '{{ faketime_pkg }}'
       register: faketime_package_installed
       until: faketime_package_installed is success
+      when: ansible_distribution != 'Alpine'
+
+    - name: install faketime packages - Alpine
+      command: apk add -U {{ faketime_pkg }} --repository=http://dl-cdn.alpinelinux.org/alpine/edge/testing
```
Comment (samdoran) [inline test/integration/targets/setup_cron/tasks/main.yml:30]: Any reason this uses `command` rather than the `apk` module?

Comment (samdoran) [reply test/integration/targets/setup_cron/tasks/main.yml:30]: Nevermind, I see the explanation in a later comment.

Comment (sivel) [reply test/integration/targets/setup_cron/tasks/main.yml:30]: The same reason I commented on at https://github.com/ansible/ansible/pull/70946#discussion_r461680298

`apk` isn't core, and I see no reason to vendor it for these simple use cases in tests.

File: test/integration/targets/loops/tasks/main.yml [hunk 6]
```diff
@@ -27,20 +27,27 @@
...
 
-    - name: ensure lag, since there is 3 rounds, and 0.5 seconds between, it should last 1.2 seconds, but allowing leeway due to CI lag
-      assert:
-        that:
-            - tdiff|float >= 1.2
-            - tdiff|int < 3
-      vars:
+    - name: Debug times var
+      debug:
+        var: times
```
Comment (samdoran) [inline test/integration/targets/loops/tasks/main.yml:37]: Do we still need this in the final test? Maybe it was just for troubleshooting?

**Review by samdoran — APPROVED** [review]
I think my only concern is we're just skipping Alpine for many tests. Do we have plans to fix some of these things to work on Alpine? I know It's a ton of work, just wondering how we'll track what needs to be done.

File: test/integration/targets/loops/tasks/main.yml (hunk 6 above)
Comment (sivel) [reply test/integration/targets/loops/tasks/main.yml:37]: I prefer adding debugs, when we do assertions on data.  Without it, you cannot directly troubleshoot the variables, without modifying the playbook.  So it wasn't accidental, it's to make it easier to troubleshoot.

**Comment by sivel** [comment]
> I think my only concern is we're just skipping Alpine for many tests. Do we have plans to fix some of these things to work on Alpine? I know It's a ton of work, just wondering how we'll track what needs to be done.

The only ones were are really skipping are:

1. service - we don't have an openrc module in core, and we're not planning to extend the generic `service` module
1. user - the user module fails miserably on the tests, and that is being worked on
1. parts of cron - this is just due to limitations in the cron implementation missing some features

I think the user module is probably the only thing we will actively work toward.

**Comment by sivel** [comment]
Ok, we're green here again.

**Review by mattclay — APPROVED** [review]

**Pull request merged**